src/super_gradients/common/deprecate.py

```diff
@@ -0,0 +1,90 @@
+import warnings
+from functools import wraps
+from typing import Optional
+from pkg_resources import parse_version
+
+import super_gradients
+
+
+def deprecated(deprecated_in_v: str, remove_in_v: str, target: Optional[callable] = None, reason: str = ""):
+    """
+    Decorator to mark a callable as deprecated. Works on functions and classes.
+    It provides a clear and actionable warning message informing
+    the user about the version in which the function was deprecated, the version in which it will be removed,
+    and guidance on how to replace it.
+
+    :param deprecated_in_v: Version number when the function was deprecated.
+    :param remove_in_v:     Version number when the function will be removed.
+    :param target:          (Optional) The new function that should be used as a replacement. If provided, it will guide the user to the updated function.
+    :param reason:          (Optional) Additional information or reason for the deprecation.
+
+    Example usage:
+        If a direct replacement function exists:
+        >> from new.module.path import new_get_local_rank
+
+        >> @deprecated(deprecated_in_v='3.2.0', remove_in_v='4.0.0', target=new_get_local_rank, reason="Replaced for optimization")
+        >> def get_local_rank():
+        >>     return new_get_local_rank()
+
+        If there's no direct replacement:
+        >> @deprecated(deprecated_in_v='3.2.0', remove_in_v='4.0.0', reason="Function is no longer needed due to XYZ reason")
+        >> def some_old_function():
+        >>     # ... function logic ...
+
+        When calling a deprecated function:
+        >> from some_module import get_local_rank
+        >> get_local_rank()
+        DeprecationWarning: Function `some_module.get_local_rank` is deprecated. Deprecated since version `3.2.0`
+        and will be removed in version `4.0.0`. Reason: `Replaced for optimization`.
+        Please update your code:
+          [-] from `some_module` import `get_local_rank`
+          [+] from `new.module.path` import `new_get_local_rank`.
+    """
+
+    def decorator(old_func: callable) -> callable:
+
+        if parse_version(super_gradients.__version__) >= parse_version(remove_in_v):
+            raise EnvironmentError(
```

Comment:
Lets change this to import error

Reply:
[bd2d58c](https://github.com/Deci-AI/super-gradients/pull/1416/commits/bd2d58c50f1906bf38f563d4c2ae46c1bcb7f8e4)